tasties_app/views.py

```diff
@@ -96,3 +97,40 @@ def view_recipe(request, recipe_id=None):
     categories = recipe.categories.all()
     context = {'recipe': recipe, 'ingredients': ingredients, 'rating': rating, 'categories': categories}
     return render(request, 'tasties_app/view_recipe.html', context)
+
+
+@login_required(login_url='login')
+def create_recipe(request):
+    recipe = Recipe(author_id=request.user)
+    IngredientFormSet = inlineformset_factory(Recipe,
+                                              Ingredient,
+                                              fields=('description', 'measurement_unit', 'amount'),
+                                              min_num=1,
+                                              validate_min=True,
+                                              extra=9)
+
+    if request.method == 'POST':
+        recipe_form = CreateRecipeForm(request.POST, request.FILES, instance=recipe)
+        ingredient_formset = IngredientFormSet(request.POST, instance=recipe)
+        if recipe_form.is_valid():
+            recipe = recipe_form.save(commit=False)
+            if ingredient_formset.is_valid():
+                recipe = recipe_form.save()
+                if not recipe.recipe_picture:
+                    recipe.recipe_picture = 'images/tasties_logo_round.png'
+                    recipe.save()
+                ingredient_formset.save()
+                return redirect(f'/view_recipe/{recipe.id}/')
+            else:
+                for ingredient_form_errors in ingredient_formset.errors:
```

Comment:
Maybe this could be done in one for loop?

Reply:
Like with list comprehension? I think readability would suffer if I tried that.

Reply:
I think what @natifridman meant is to a single for-loop, like we did in line 83 in the register method.

Reply:
> I think what @natifridman meant is to a single for-loop, like we did in line 83 in the register method.

Please provide a link to what you want me to see...

Reply:
```suggestion
                for error_message in form.errors.values():
                     messages.error(request, error_message)
```

It's just those lines from `register` method in `views.py`

Reply:
> It's just those lines from `register` method in `views.py`

The difference is that I'm iterating over a `formset`, which has many forms.